HHS/OPRE-OPS
# OPS-970 HHS Provider, MultiAuth, FakeAuth, User Edit

## What changed

* Wired up HHS Authentication Provider
* Created `MultiAuth` Login schema
* Created `FakeAuth` Login for offline / dev / demo login
* Created User Edit page, mostly for `new user` profile confirmation
* Updated the `docker-compose` so that `backend` waits for `data-import` to comlete
* E2E no longer requires a different `docker-compose.e2e.yml` stack to be running
  * This will still require local DB cleanup, or re-running of `data-import` manually if run consecutively, since not all E2E tests do a full cleanup 😞 

## Issue

#970 
#963 
#1206 

## How to test

### Test FakeAuth
1. Visit the homepage, you should be redirected to /login
2. Click "Sign in with FakeAuth"
3. Click "Admin User"
4. You should be logged in as "admin.demo@email.com"

### Test AMS Login (Requires PIV)
1. Click "Sign in with HHS AMS"
2. Should be redirected to AMS login
3. Login with PIV
4. Should be redirected back to the homepage, logged in as your user.

## Screenshots

![image](https://github.com/HHS/OPRE-OPS/assets/56687505/e8cf0956-cfa4-4fcc-bdbf-91523c5629c3)

![image](https://github.com/HHS/OPRE-OPS/assets/56687505/57e1c6ae-1e54-49ca-a8f8-bd8527521271)

![image](https://github.com/HHS/OPRE-OPS/assets/56687505/851ba51f-121b-4832-a2c1-585e58c2da17)

## Links

TBD
